tree.go

```diff
@@ -0,0 +1,41 @@
+package fastmerkle
+
+// Node represents a single node in the Merkle
+type Node struct {
+	Hash []byte // The hash of the children nodes
+
+	Left   *Node // Reference to the left child
+	Right  *Node // Reference to the right child
+	Parent *Node // Reference to the parent node
+}
+
+// duplicate creates a copy of the node
+// and its reference peers
+func (n *Node) duplicate() *Node {
+	return &Node{
+		Hash:   n.Hash,
+		Left:   n.Left,
+		Right:  n.Right,
+		Parent: n.Parent,
+	}
+}
+
+// GetHash returns the hash of the node's children
+func (n *Node) GetHash() []byte {
```

Comment:
Why have a getter if the field is public ?

Reply:
You're right - we don't need to expose these fields.

Fixed 👍 

[09a2b13](https://github.com/Trapesys/fastmerkle/pull/1/commits/09a2b1366526686e0fd719ad6b1610e1ca4c34d7)

Reply:
Additional 👍 

[23a563d](https://github.com/Trapesys/fastmerkle/pull/1/commits/23a563d7ea75b5578450a8259f582b3ae2d95c71)